src/notebooks/lms_utilities.py

```diff
@@ -1,158 +1,170 @@
+        return _read_csv(file)
+
+    return pd.DataFrame()
+
+def _get_data_for_section(base_directory: str, sections: pd.DataFrame, callback: Callable) -> pd.DataFrame:
+    df = pd.DataFrame()
+
+    if sections.empty:
+        print("No sections have been loaded, therefore no section sub-files can be read.")
+        return
+
+    for _, section_id in sections[["SourceSystemIdentifier"]].itertuples():
+        sa = callback(base_directory, section_id)
+
+        if not sa.empty:
+            df = df.append(sa)
+
+    return df
+
+def get_all_section_associations(base_directory: str, sections: pd.DataFrame) -> pd.DataFrame:
+    return _get_data_for_section(base_directory, sections, get_section_associations)
+
+def get_user_activities(base_directory: str, section_id: int) -> pd.DataFrame:
+    file = get_user_activities_file(base_directory, section_id)
+
+    if file is not None:
+        return _read_csv(file)
+
+    return pd.DataFrame()
+
+def get_all_user_activities(base_directory: str, sections: pd.DataFrame) -> pd.DataFrame:
+    return _get_data_for_section(base_directory, sections, get_user_activities)
+
+def get_assignments(base_directory: str, section_id: int) -> pd.DataFrame:
+    file = get_assignments_file(base_directory, section_id)
+
+    if file is not None:
+        return _read_csv(file)
+
+    return pd.DataFrame()
+
+def get_all_assignments(base_directory: str, sections: pd.DataFrame) -> pd.DataFrame:
+    return _get_data_for_section(base_directory, sections, get_assignments)
+
+def get_submissions(base_directory: str, section_id: int, assignment_id: int) -> pd.DataFrame:
+    file = get_submissions_file(base_directory, section_id, assignment_id)
+
+    if file is not None:
+        return _read_csv(file)
+
+    return pd.DataFrame()
+
+def get_all_submissions(base_directory: str, assignments: pd.DataFrame) -> pd.DataFrame:
+    df = pd.DataFrame()
+    columns = ["SourceSystemIdentifier", "LMSSectionSourceSystemIdentifier"]
+    for _, assignment_id, section_id  in assignments[columns].itertuples():
+        s = get_submissions(base_directory, section_id, assignment_id)
+
+        if not s.empty:
+            df = df.append(s)
+
+    return df
+
+def get_grades(base_directory: str, section_id: int) -> pd.DataFrame:
+    file = get_grades_file(base_directory, section_id)
+
+    if file is not None:
+        return _read_csv(file)
+
+    return pd.DataFrame()
+
+def get_all_grades(base_directory: str, sections: pd.DataFrame) -> pd.DataFrame:
+    return _get_data_for_section(base_directory, sections, get_grades)
+
+def get_attendance_events(base_directory: str, section_id: int) -> pd.DataFrame:
+    file = get_attendance_events_file(base_directory, section_id)
+
+    if file is not None:
+        return _read_csv(file)
+
+    return pd.DataFrame()
+
+def get_all_attendance_events(base_directory: str, sections: pd.DataFrame) -> pd.DataFrame:
+    return _get_data_for_section(base_directory, sections, get_attendance_events)
```

Comment:
This file no longer exists; you need to merge the `main` branch into your branch.

Please see the latest [filesystem tutorial](https://github.com/Ed-Fi-Exchange-OSS/Ed-Fi-X-Fizz/blob/main/src/notebooks/filesystem-tutorial.ipynb), which has code like this:

```python
# Load this helper module to access the functions
# Instead of loading from a published package, we're going to hack the 
# module search path to find the package on the local file system.
import sys
sys.path.append(os.path.join("..","file-utils"))

from lms_file_utils import file_repository # contains functions that get directory paths
from lms_file_utils import file_reader # contains the functions that load data frames
```

And now a function like `get_all_attendance_events` needs to be called with `file_reader.get_all_attendance_events`.